ToolBox/CoinApi/CoinApiDataQueueHandler.cs

```diff
@@ -0,0 +1,360 @@
+                _subscribedSymbols = _subscribedSymbols.Where(x => !symbolsToUnsubscribe.Contains(x)).ToHashSet();
+
+                ProcessSubscriptionRequest();
+            }
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            _connectionHandler.DisposeSafely();
+
+            if (_webSocket.IsOpen)
+            {
+                _webSocket.Close();
+            }
+        }
+
+        /// <summary>
+        /// Helper method used in QC backend
+        /// </summary>
+        /// <param name="markets">List of LEAN markets (exchanges) to subscribe</param>
+        public void SubscribeMarkets(List<string> markets)
+        {
+            if (markets.Count == 0) return;
+
+            Log.Trace($"CoinApiDataQueueHandler.SubscribeMarkets(): {string.Join(",", markets)}");
+
+            _subscribedExchanges = markets.ToList();
+
+            var message = JsonConvert.SerializeObject(new HelloMessage
+            {
+                ApiKey = _apiKey,
+                Heartbeat = true,
+                SubscribeDataType = new[] { "trade", "quote" },
+                SubscribeFilterSymbolId = markets.Select(x => _symbolMapper.GetExchangeId(x)).ToArray()
+            });
+
+            _webSocket.Send(message);
+
+            _connectionHandler.EnableMonitoring(true);
+        }
+
+        private void ProcessSubscriptionRequest()
+        {
+            if (_subscriptionsPending) return;
+
+            _lastSubscribeRequestUtcTime = DateTime.UtcNow;
+            _subscriptionsPending = true;
+
+            Task.Run(() =>
+            {
+                while (true)
+                {
+                    DateTime requestTime;
+                    List<Symbol> symbolsToSubscribe;
+                    lock (_lockerSubscriptions)
+                    {
+                        requestTime = _lastSubscribeRequestUtcTime.Add(_subscribeDelay);
+                        symbolsToSubscribe = _subscribedSymbols.ToList();
+                    }
+
+                    if (DateTime.UtcNow > requestTime)
+                    {
+                        // minimum delay has passed since last subscribe request, send the subscribe request
+                        SubscribeSymbols(symbolsToSubscribe);
+
+                        lock (_lockerSubscriptions)
+                        {
+                            _lastSubscribeRequestUtcTime = DateTime.UtcNow;
+                            if (_subscribedSymbols.Count == symbolsToSubscribe.Count)
+                            {
+                                // no more subscriptions pending, task finished
+                                _subscriptionsPending = false;
+                                break;
+                            }
+                        }
+                    }
+
+                    Thread.Sleep(200);
```

Comment:
I think we might be able to sleep the exact amount we need, so `DateTime.UtcNow > requestTime` is true?

Reply:
The 200ms is not related to the 250ms delay mentioned above, this value was chosen only to minimize CPU usage in this loop.

Reply:
This loop is intended to only run until we've processed all the subscriptions, correct? Also, the only reason we _wouldn't_ be able to process all of the subscriptionsis due to the 5 second change via `DateTime.UtcNow > requestTime` -- or perhaps I'm missing how this is intended to operate. If the purpose of the loop is to continually check until the required 5 seconds has passed, then I agree w/ Martin that we could sleep (or delay, see other comment) until the required amount of time has passed instead of napping for ~200ms only to wake up and take another nap because of the the request time checks.

Reply:
The purpose of this task is to group subscription requests together because we can't send Hello messages before 5 seconds have passed since the previous message. We don't want to wait 5 seconds after the last subscription request is received so we submit the request if at least 250ms has elapsed since the last request.

Note: the check for 5-second minimum elapsed time between Hello messages is not implemented yet. 
I'm working on it now and will also change the logic to wait the correct amount of time.